src/components/ObsDetails/ObsDetails.js

```diff
@@ -70,31 +71,84 @@ const ObsDetails = ( ): Node => {
   const bottomSheetModalRef = useRef( null );
   const [addingComment, setAddingComment] = useState( false );
   const [snapPoint, setSnapPoint] = useState( 100 );
-  const apiToken = useApiToken( );
-  const [localObservation, setLocalObservation] = useState( null );
+  const realm = useRealm( );
+  const localObservation = realm?.objectForPrimaryKey( "Observation", uuid );
 
   const queryClient = useQueryClient( );
 
   const {
-    data: remoteObservation
+    data: remoteObservation,
+    refetch: refetchRemoteObservation
   } = useAuthenticatedQuery(
     ["fetchRemoteObservation", uuid],
     optsWithAuth => fetchRemoteObservation( uuid, { }, optsWithAuth )
   );
 
-  const realm = useRealm( );
+  const observation = localObservation || remoteObservation;
 
-  useEffect( ( ) => {
-    setLocalObservation( realm?.objectForPrimaryKey( "Observation", uuid ) );
-  }, [realm, uuid] );
+  const handleSuccess = {
+    onSuccess: ( ) => {
+      queryClient.invalidateQueries( ["fetchRemoteObservation", uuid] );
+      refetchRemoteObservation( );
+    }
+  };
 
-  const observation = localObservation || remoteObservation;
+  const createCommentMutation = useAuthenticatedMutation(
+    ( body, optsWithAuth ) => createComments( {
+      comment: {
+        body,
+        parent_id: uuid,
+        parent_type: "Observation"
+      }
+    }, optsWithAuth ),
+    handleSuccess
+  );
+
+  const handleIdentificationMutation = {
+    onSuccess: data => setIds( [...ids, data[0]] ),
+    onError: e => {
+      let error = null;
+      if ( e ) {
+        error = t( "Couldnt-create-identification", { error: e.message } );
+      } else {
+        error = t( "Couldnt-create-identification", { error: t( "Unknown-error" ) } );
+      }
+
+      // Remove temporary ID and show error
+      setIds( [...ids] );
+      Alert.alert(
+        "Error",
+        error,
+        [{ text: t( "OK" ) }],
+        {
+          cancelable: true
+        }
+      );
+    }
+  };
```

Comment:
Could you (or should you) refactor this to use `optimistic updates`? Maybe you could just get rid of `onIDAdded()`. Maybe a subject for later work.

Reply:
Optimistic updates as in show the user the regular ID result with black text instead of the ghost result with gray text? I'm down to do that, but was trying to keep functionality the same as it was before in this PR.

Reply:
Meant to include a link. Was thinking about https://react-query-v3.tanstack.com/guides/optimistic-updates. Happy to just maintain prior functionality though.

Reply:
This might be good to explore in a separate issue. I poked around at this for about an hour, but couldn't immediately get it working for updating a comment.